src/vs/editor/contrib/parameterHints/parameterHintsWidget.ts

```diff
@@ -312,9 +312,11 @@ export class ParameterHintsWidget extends Disposable implements IContentWidget {
 		} else if (Array.isArray(param.label)) {
 			return param.label;
 		} else {
-			const idx = signature.label.lastIndexOf(param.label);
+			const regex = new RegExp(`\\b${param.label}\\b`, 'g');
```

Comment:
In case the `param.label` contains special regular expression, make sure to use [`escapeRegExpCharacters`](https://github.com/microsoft/vscode/blob/25ad7485b3aa2dea43ab945b4f6af94cf0537b09/src/vs/base/common/strings.ts#L68)